[Traefik Plugin Catalog] Plugin Analyzer has detected a problem.

The plugin was not imported into Traefik Plugin Catalog.

Cause:
```
failed to run the plugin with Yaegi: the load of the plugin takes too much time(10s), or an error, inside the plugin, occurs during the load: 1:21: import "github.com/tomMoulard/fail2ban" error: /tmp/traefik-plugin-gop1901121605/src/github.com/tomMoulard/fail2ban/fail2ban.go:17:2: import "github.com/tomMoulard/fail2ban/pkg/response/status" error: /tmp/traefik-plugin-gop1901121605/src/github.com/tomMoulard/fail2ban/pkg/response/status/http_code_range.go:5:2: import "slices" error: unable to find source related to: "slices"
```
Traefik Plugin Analyzer will restart when you will close this issue.

If you believe there is a problem with the Analyzer or this issue is the result of a false positive, please fill an issue on [piceus](https://github.com/traefik/piceus) repository.

--- a/pkg/response/status/http_code_range.go
+++ b/pkg/response/status/http_code_range.go
@@ -2,7 +2,6 @@ package status
 
 import (
 	"fmt"
-	"slices"
 	"strconv"
 	"strings"
 )
@@ -43,7 +42,11 @@ func NewHTTPCodeRanges(strBlocks []string) (HTTPCodeRanges, error) {
 
 // Contains tests whether the passed status code is within one of its HTTP code ranges.
 func (h HTTPCodeRanges) Contains(statusCode int) bool {
-	return slices.ContainsFunc(h, func(block [2]int) bool {
-		return statusCode >= block[0] && statusCode <= block[1]
-	})
+	for ranges := range h {
+		if statusCode >= h[ranges][0] && statusCode <= h[ranges][1] {
+			return true
+		}
+	}
+
+	return false
 }

--- a/pkg/response/status/http_code_range_test.go
+++ b/pkg/response/status/http_code_range_test.go
@@ -1,0 +1,51 @@
+package status
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestContains(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name       string
+		ranges     HTTPCodeRanges
+		statusCode int
+		expected   assert.BoolAssertionFunc
+	}{
+		{
+			name:       "empty",
+			ranges:     HTTPCodeRanges{},
+			statusCode: 200,
+			expected:   assert.False,
+		},
+		{
+			name:       "single",
+			ranges:     HTTPCodeRanges{{200, 200}},
+			statusCode: 200,
+			expected:   assert.True,
+		},
+		{
+			name:       "single out of range high",
+			ranges:     HTTPCodeRanges{{200, 200}},
+			statusCode: 201,
+			expected:   assert.False,
+		},
+		{
+			name:       "single out of range low",
+			ranges:     HTTPCodeRanges{{200, 200}},
+			statusCode: 199,
+			expected:   assert.False,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			test.expected(t, test.ranges.Contains(test.statusCode))
+		})
+	}
+}

